Userland/Services/FileOperation/main.cpp

```diff
@@ -365,3 +369,41 @@ int execute_work_items(Vector<WorkItem> const& items)
     outln("FINISH");
     return 0;
 }
+
+AK::Result<AK::NonnullRefPtr<Core::File>, AK::OSError> open_destination_file(String const& destination)
+{
+    auto destination_file_or_error = Core::File::open(destination, (Core::OpenMode)(Core::OpenMode::WriteOnly | Core::OpenMode::Truncate | Core::OpenMode::MustBeNew));
+    if (destination_file_or_error.is_error() && destination_file_or_error.error().error() == EEXIST) {
+        return open_destination_file(deduplicate_destination_file_name(destination));
+    }
+    return destination_file_or_error;
+}
+
+String deduplicate_destination_file_name(String const& destination)
+{
+    LexicalPath destination_path(destination);
+    auto title_without_counter = destination_path.title();
+    size_t next_counter = 1;
+
+    auto last_space_index = title_without_counter.find_last(' ');
+    if (last_space_index.has_value()) {
+        auto counter_suffix = title_without_counter.substring_view(*last_space_index + 1);
+        if (counter_suffix.starts_with('(') && counter_suffix.ends_with(')')) {
+            auto counter_string = counter_suffix.substring_view(1, counter_suffix.length() - 2);
+            auto last_counter = counter_string.to_uint();
+            if (last_counter.has_value()) {
+                next_counter = *last_counter + 1;
+                title_without_counter = title_without_counter.substring_view(0, *last_space_index);
+            }
+        }
+    }
+
+    StringBuilder basename;
+    basename.append(String::formatted("{} ({})", title_without_counter, next_counter));
```

Comment:
:thinking: could we come up with a nicer naming scheme here? `foo (x)` is like the ultimate anti-nice-shell-ux file name that could exist, in that you need at least three backslashes to use it...

Reply:
I usually prefer `file-1` instead of `file (1)`, much easier to type, no backslashes or even a shift (if you use an underscore) needed.

Reply:
Using `file-1` sounds like a great idea! Updated my change.